app/controllers/application_controller.rb

```diff
@@ -182,118 +182,123 @@ def setup_body_classes
-      elsif url_check[2] == "health"
-        title = "Health Summary Dashboard"
-      elsif url_check[2] == "notebooks"
-        title = "Notebook Summary Dashboard"
-      elsif url_check[2] == "notebook_similarity"
-        title = "Notebook Similarity Dashboard"
-      elsif url_check[2] == "packages"
-        title = "Package Usage Dashboard"
-      elsif url_check[2] == "recommender_summary"
-        title = "Recommendation Summary Dashboard"
-      elsif url_check[2] == "trendiness"
-        title = "Trendiness Summary Dashboard"
-      elsif url_check[2] == "user_similarity"
-        title = "User Similarity Dashboard"
-      elsif url_check[2] == "user_summary"
-        title = "User Summary Dashboard"
-      elsif url_check[2] == "warning"
-        title = "Admin Site Banner"
+    browser = Browser.new(request.env["HTTP_USER_AGENT"])
+    if !browser.modern?
+      title = "NBGallery"
+    else
+      url_check = request.path.split("/")
+      url = request.path.sub("/","").titlecase.sub("/",": ").gsub("/"," ").gsub("_"," ").gsub(/\d+-/, "")
+      if url_check[1] == "change_requests"
+        if url_check[2] == nil
+          title = "Change Requests"
+        elsif url_check[2] == "all"
+          title = "All Change Requests"
+        else
+          title = "Change Request for \"#{@notebook.title}\""
+        end
+      elsif url_check[1] == "reviews"
+        if url_check[2] == nil
+          title = "All Reviews"
+        elsif defined? @notebook.title && defined? @review.revtype
+          title = "#{GalleryConfig.reviews[@review.revtype].label.capitalize} Review of \"#{@notebook.title}\""
+        else
+          title = "#{url}"
+        end
+      elsif defined? @notebook.title
+        if url_check[3] == "metrics"
+          title = "Metrics of \"#{@notebook.title}\""
+        elsif url_check[3] == "revisions"
+          title = "Revisions of \"#{@notebook.title}\""
+        elsif url_check[3] == "reviews"
+          title = "Reviews of \"#{@notebook.title}\""
+        elsif url_check[3] == "code_cells" && url_check[4] != nil
+          title = "Code Cell #{url_check[4]} of \"#{@notebook.title}\""
+        else
+          title = "#{@notebook.title}"
+        end
+      elsif url_check[1] == "notebooks" && params[:q] != nil && params[:q].length > 0
+        title = "Search for \"#{params[:q]}\""
+      elsif url_check[1] == "notebooks"
+        if url_check[2] == nil
+          title = "All Notebooks"
+        elsif url_check[2] == "stars"
+          title = "Starred Notebooks"
+        elsif url_check[2] == "recommended"
+          title = "Recommended for Me"
+        elsif url_check[2] == "recently_executed"
+          title = "Notebooks Recently Executed"
+        elsif url_check[2] == "shared_with_me"
+          title = "Notebooks Shared with Me"
+        else
+          title = "#{url}"
+        end
+      elsif url_check[1] == "groups"
+        if url_check[2] == nil
+          title = "All Groups"
```

Comment:
What if the group doesn't exist?  Can you still get here?  What about
if(@group != nil)
   title= Group group.name
elsif url_check[2]=nil
  All Groups
else
  title="Groups"
end if

Reply:
I tried to hit all of the endpoints for notebooks, users, groups, and change requests but for every error, the only time the element is not attached is when the browser is not modern.